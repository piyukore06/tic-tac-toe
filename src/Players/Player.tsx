import React from 'react';
import { PlayerType } from '../App';

const Player = (props: { hasWon: boolean, playerNo: PlayerType, onNewGame: () => void }) => {
  return (
    <div className="player">
        Player {props.playerNo} {props.hasWon && "ðŸŽ‰ðŸ¥³"}
        {props.hasWon && <button onClick={props.onNewGame}>New Game</button>}

    </div>
  );
}

export default Player;
