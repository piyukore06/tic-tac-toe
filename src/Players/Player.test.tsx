import React from 'react';
import renderer from 'react-test-renderer';
import Player from './Player';

test('renders proper hasWon output', () => {
  const component1 = renderer.create(<Player hasWon={false} playerNo={'X'}  onNewGame={() => {}} />);
  let tree1 = component1.toJSON();
  expect(tree1).toMatchSnapshot();

  const component2 = renderer.create(<Player hasWon={true} playerNo={'X'}  onNewGame={() => {}} />);
  let tree2 = component2.toJSON();
  expect(tree2).toMatchSnapshot();


  const component3 = renderer.create(<Player hasWon={'NOONE'} playerNo={'X'}  onNewGame={() => {}} />);
  let tree3 = component3.toJSON();
  expect(tree3).toMatchSnapshot();
});


test('renders proper playerNo output', () => {
    const component1 = renderer.create(<Player hasWon={false} playerNo={'X'}  onNewGame={() => {}} />);
    let tree1 = component1.toJSON();
    expect(tree1).toMatchSnapshot();
  
    const component2 = renderer.create(<Player hasWon={false} playerNo={'O'}  onNewGame={() => {}} />);
    let tree2 = component2.toJSON();
    expect(tree2).toMatchSnapshot();
});