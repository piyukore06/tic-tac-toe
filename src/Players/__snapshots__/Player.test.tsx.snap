// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders proper hasWon output 1`] = `
<div
  className="player"
>
  Player 
  X
  <br />
  <br />
</div>
`;

exports[`renders proper hasWon output 2`] = `
<div
  className="player"
>
  Player 
  X
  <br />
  ðŸŽ‰ðŸ¥³
  <br />
  <button
    onClick={[Function]}
  >
    New Game
  </button>
</div>
`;

exports[`renders proper hasWon output 3`] = `
<div
  className="player"
>
  Player 
  X
  <br />
  No one won, Try Again!!
  <br />
  <button
    onClick={[Function]}
  >
    New Game
  </button>
</div>
`;

exports[`renders proper playerNo output 1`] = `
<div
  className="player"
>
  Player 
  X
  <br />
  <br />
</div>
`;

exports[`renders proper playerNo output 2`] = `
<div
  className="player"
>
  Player 
  O
  <br />
  <br />
</div>
`;
